robot_smach_states/src/robot_smach_states/utility.py

```diff
@@ -368,25 +368,29 @@ def execute(self, userdata=None):
 
 
 class LockDesignator(smach.State):
-    def __init__(self, locking_designator):
+    def __init__(self, locking_designator, resolve_print=True):
```

Comment:
I dont like this as an argument. Like this you get functions with 10+ flag arguments to trigger different behaviours. And this one just changes wether something gets printed.
I say just print it every time.

Reply:
The problem with the print is that it causes the designator to be resolved. While you only want the designator to be resolved when you use it. This isn't a problem for a normal designator, but is is for a locking designator.